### ISSUE TITLE
Improve code coverage

### ISSUE THREAD

**Issue by ksubramz** [CppMicroServices/CppMicroServices#235]
This is an umbrella issue to track the work done to improve the code coverage of our test suite. (Landing page: https://codecov.io/gh/cppmicroservices/CppMicroServices)

**Comment by rasane** [comment]
Hello, I Would like to help.. I added 1 test and confirmed it ran successfully and code path was executed. I Wanted to confirm if it affects code metrics, how can I do that? 

Apart from raising a pr and running code coverage at about link, how can I set up on my personal branch?
Ps: haven't forked officially yet as i m trying to understand my capability to contribute positively. Let me know if I missed some obvious note, apologies in advance.

Thanks, rasane

**Comment by jeffdiclemente** [comment]
Hi rasane,

Thanks for writing a test!

There are a few ways you could determine whether the test affects code metrics...
### Manual Inspection
Use the codecov.io link above to navigate to the source file(s) your test uses. Visually inspect which lines in the source aren't hit by a test and then check your test code to see if it walks any of those code paths. If it does, you can be sure that the test improves code coverage.

### Run gcov and lcov locally on your machine
In a nutshell, gcov and lcov are used to gather and create the code coverage data. You can always run these manually on a Linux box.

### Fork CppMicroServices and send a PR
This is typically very quick and easy and doesn't require running tools manually.

**Comment by rasane** [comment]
Thanks for the detailed explanation. I don't have a Linux box at the moment, will install one this weekend.. I haven't used gcov/lcov before, sound like handy tools..
